lib/awful/layout/suit/tile.lua.in

```diff
@@ -156,26 +155,28 @@ local function tile_group(cls, wa, orientation, fact, group)
     size = math.min(size, available)
 
     local coord = wa[y]
-    local geom = {}
     local used_size = 0
     local unused = wa[height]
     for c = group.first,group.last do
         local i = c - group.first +1
-        geom[width] = size - cls[c].border_width * 2
-        geom[height] = math.floor(unused * fact[i] / total_fact) - cls[c].border_width * 2
+        local geom = {}
+        geom[width] = size
+        geom[height] = math.floor(unused * fact[i] / total_fact)
         geom[x] = group.coord
         geom[y] = coord
-        geom = cls[c]:geometry(geom)
```

Comment:
I just noticed that this is a regression. The tiled layout takes into account differences between the geometry you tell the client to use and the actual resulting geometry after it is set. This is no problem if you aren't honoring size hints, but otherwise it can result in weird gaps between clients.

Is it possible to figure out the resulting geometry without trying to set it? Otherwise flickering will be unavoidable on the tiled layout due to the two passes.